src/main/java/org/kiwiproject/dropwizard/error/job/CleanupApplicationErrorsJob.java

```diff
@@ -0,0 +1,38 @@
+package org.kiwiproject.dropwizard.error.job;
+
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.dropwizard.error.config.CleanupConfig;
+import org.kiwiproject.dropwizard.error.dao.ApplicationErrorDao;
+
+import java.time.ZonedDateTime;
+
+/**
+ * Job that can be configured to run on a regular interval that will delete expired application errors.
+ *
+ * @see CleanupConfig for options on what will be deleted and when
+ */
+@Slf4j
+public class CleanupApplicationErrorsJob implements Runnable {
+
+    private final CleanupConfig config;
+    private final ApplicationErrorDao errorDao;
+
+    public CleanupApplicationErrorsJob(CleanupConfig config, ApplicationErrorDao errorDao) {
```

Comment:
When we build an `ErrorContext` using `ErrorContextBuilder`, we (by default) also setup a health check. Shouldn't we also set up the cleanup job? We could also provide options to specify a `CleanupConfig` or to skip setting up the cleanup job entirely, in the same manner that we we allow customization of the health check using `timeWindow(TimeWindow)` and `skipHealthCheck()`.

For example, with a custom `CleanupConfig`:

```java
var errorContext = ErrorContextBuilder.newInstance()
    .environment(theDropwizardEnvironment)
    .serviceDetails(serviceDetails)
    .cleanupConfig(cleanupConfig)
    .buildWithJdbi3(jdbi);
```

Or, to skip adding the cleanup job:

```java
var errorContext = ErrorContextBuilder.newInstance()
    .environment(theDropwizardEnvironment)
    .serviceDetails(serviceDetails)
    .skipCleanup()
    .buildWithJdbi3(jdbi);
```

Since we require the Dropwizard `Environment`, we can easily obtain the `LifecycleEnvironment` in order to create a `ScheduledExecutorService`.

We should also strongly consider having options in `CleanupConfig` for the scheduled executor, e.g. how often to execute, initial delay, thread name format, shutdown time, etc.

Reply:
Working on this now